### PR TITLE
Closes #4905 Minimalist blogger thirdparty class for delay JS compatibility

### LINKED ISSUE #4905: Delay JavaScript Execution compatibility with the MinimalistBlogger
**Before submitting an issue please check that you’ve completed the following steps:**
- [x] Made sure you’re on the latest version `3.11.0.2`
- [x] Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
When Delay JavaScript Execution is using with the theme [MinimalistBlogger](https://superbthemes.com/minimalistblogger/), the following error shows on the browser console on user interaction: 

```
Uncaught TypeError: Cannot read properties of undefined (reading 'on')
    at script.js?ver=1649406382:4:2030
    at script.js?ver=1649406382:4:2096
(anonymous) @ script.js?ver=1649406382:4
(anonymous) @ script.js?ver=1649406382:4
```

Referring to the following file: 

```
/themes/minimalistblogger/js/script.js
```

The issue only happens when the `jquery` library is not excluded. It's fixed once it's excluded. 

**To Reproduce**
1. Use the MinimalistBlogger theme
2. Enable Delay JavaScript Execution 
3. See error

**Expected behavior**
MinimalistBlogger should work out of the box with Delay JavaScript Execution. 

**Screenshots**
N/A

**Additional context**
Ticket - https://secure.helpscout.net/conversation/1839489447/336073

**Backlog Grooming (for WP Media dev team use only)**
- [x] Reproduce the problem
- [x] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description

Fix a problem with jQuery in Minimalist Blogger when Delay JS is activated by excluding jQuery from delayed scripts when the theme is active.

Fixes #4905

## Type of change

Please delete options that are not relevant.

- [ ] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

No

## How Has This Been Tested?

- [x] Automated tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by CrochetFeve0251** [comment]
@Tabrisrp @piotrbak the cleaned branche

File: inc/ThirdParty/Themes/ThirdpartyTheme.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	 * @var string
+	 */
+	protected static $theme_name = '';
+	/**
+	 * Check if the theme or one of its child theme is activated.
+	 *
+	 * @param WP_Theme $theme current theme.
+	 * @return bool
+	 */
+	protected static function is_theme( $theme = null ) {
```
Comment (remyperona) [inline inc/ThirdParty/Themes/ThirdpartyTheme.php]:
```suggestion
	protected static function is_current_theme( $theme = null ) {
```

File: inc/ThirdParty/Themes/MinimalistBlogger.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	 */
+	protected static $theme_name = 'minimalist-blogger';
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		if ( ! self::is_theme() ) {
```
Comment (remyperona) [inline inc/ThirdParty/Themes/MinimalistBlogger.php]:
```suggestion
		if ( ! self::is_current_theme() ) {
```

**Review by remyperona — CHANGES_REQUESTED** [review]
One change to make the method name more explicit

**Review by remyperona — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@CrochetFeve0251 can still see the error on newlabs using the theme https://pcm.wordpress.org/themes/minimalistblogger/. can you please check?
Screenshot from 2022-09-07 08-41-19

**Comment by piotrbak** [comment]
@Mai-Saad I'm not able to reproduce the problem. Excluding jQuery resolves the problem for me here:
https://rocketlabsqa.ovh

**Comment by Mai-Saad** [comment]
Working fine now

File: inc/ThirdParty/Themes/ThirdpartyTheme.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Themes;
+
+use WP_Theme;
+
```
Comment (jeawhanlee) [inline inc/ThirdParty/Themes/ThirdpartyTheme.php]:
```suggestion
use WP_Rocket\Event_Management\Subscriber_Interface;
```

File: inc/ThirdParty/Themes/ThirdpartyTheme.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+abstract class ThirdpartyTheme implements \WP_Rocket\Event_Management\Subscriber_Interface {
```
Comment (jeawhanlee) [inline inc/ThirdParty/Themes/ThirdpartyTheme.php]:
```suggestion
abstract class ThirdpartyTheme implements Subscriber_Interface {
```

**Review by jeawhanlee — COMMENTED** [review]
Small change

**Review by jeawhanlee — APPROVED** [review]

**Comment by vmanthos** [comment]
@CrochetFeve0251 I retested this after the latest changes and it's working as expected.

Please merge it whenever you can.

**Pull request merged**